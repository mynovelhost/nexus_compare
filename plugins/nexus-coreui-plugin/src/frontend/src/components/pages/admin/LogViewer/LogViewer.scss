/**
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/**
 * @since 3.24
 */

@import "~nexus-ui-plugin/dist/styles/colors";

.nxrm-log-viewer {
  height: 100%;
  width: 100%;

  > div {
    height: 100%;
  }

  textarea {
    background-color: $white;
    color: $nx-text-color-light;
    font-family: 'Courier New', Courier, monospace;
    font-size: 13px;
    resize: none;
    width: 100%;
    height: 96%;
  }
}

.nxrm-select-action {
  display: inline-block;
  padding-left: 10px;
  
  select {
    padding-right: 20px;
  }
}

.nxrm-log-viewer-mark-text {
    width: 200px;
    display: inline-block;
}
