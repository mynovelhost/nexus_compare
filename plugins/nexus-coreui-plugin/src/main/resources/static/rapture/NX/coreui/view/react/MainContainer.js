/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
