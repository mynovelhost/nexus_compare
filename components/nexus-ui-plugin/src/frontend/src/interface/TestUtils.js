/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {act} from 'react-dom/test-utils';
import {render} from '@testing-library/react';

/**
 * @since 3.24
 */
export default class Utils {
  static render(view, extraSelectors) {
    const selectors = render(view);
    const {queryByText} = selectors;
    return {
      ...selectors,
      loadingMask: () => queryByText('Loadingâ€¦'),
      ...extraSelectors(selectors)};
  }
}
