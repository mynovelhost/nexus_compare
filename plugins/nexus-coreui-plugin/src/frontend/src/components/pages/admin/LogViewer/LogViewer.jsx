/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React, {useEffect, useRef, useState} from 'react';
import classNames from 'classnames';
import {useMachine} from '@xstate/react';

import {
  Button,
  ContentBody,
  ExtJS,
  NxFilterInput,
  NxFontAwesomeIcon,
  Select,
  Section,
  SectionActions,
  SectionHeader,
  Textfield,
  Textarea
} from 'nexus-ui-plugin';

import {faDownload, faMapMarker} from '@fortawesome/free-solid-svg-icons';

import LogViewerMachine from "./LogViewerMachine";
import UIStrings from "../../../../constants/UIStrings";
import './LogViewer.scss';

export default function LogViewer() {
  const refreshPeriods = [
    [0,   UIStrings.LOG_VIEWER.REFRESH.MANUAL_ITEM],
    [20,  UIStrings.LOG_VIEWER.REFRESH.TWENTY_SECONDS_ITEM],
    [60,  UIStrings.LOG_VIEWER.REFRESH.MINUTE_ITEM],
    [120, UIStrings.LOG_VIEWER.REFRESH.TWO_MINUTES_ITEM],
    [300, UIStrings.LOG_VIEWER.REFRESH.FIVE_MINUTES_ITEM]
  ];
  const refreshSizes = [
    [25,  UIStrings.LOG_VIEWER.SIZE.LAST25KB_ITEM],
    [50,  UIStrings.LOG_VIEWER.SIZE.LAST50KB_ITEM],
    [100, UIStrings.LOG_VIEWER.SIZE.LAST100KB_ITEM]
  ];
  const [current, send] = useMachine(LogViewerMachine, {devTools: true});
  const selectedPeriod = current.context.period;
  const selectedSize = current.context.size;
  const mark = current.context.mark;
  const logText = current.context.data;
  const textarea = useRef(null);

  // scroll to bottom of text area
  useEffect(() => {
    if (textarea.current) {
      textarea.current.scrollTop = textarea.current.scrollHeight;
    }
  });

  function onChangePeriod({target}) {
    const newPeriod = Number(target.value);
    if (newPeriod == 0) {
      send('MANUAL_REFRESH', {period: newPeriod});
    }
    else {
      send('UPDATE_PERIOD', {period: newPeriod});
    }
  }

  function onChangeSize({target}) {
    const newSize = Number(target.value);
    send('UPDATE_SIZE', {size: newSize});
  }

  function download() {
    const url = ExtJS.urlOf('service/rest/internal/logging/log');
    ExtJS.downloadUrl(url);
  }

  function updateMark({target}) {
    send('UPDATE_MARK', {mark: target.value});
  }

  function onMarkKeyPress(event) {
    if (event.key === 'Enter') {
      event.preventDefault();
      send('INSERT_MARK');
    }
  }  

  function insertMark() {
    send('INSERT_MARK');
  }

  return <ContentBody className="nxrm-log-viewer">
    <Section>
      <SectionActions>
        <label htmlFor="period">{UIStrings.LOG_VIEWER.REFRESH.TEXT}</label>
        <SelectAction>
          <Select name="period" id="period" value={selectedPeriod} onChange={onChangePeriod}>
            {refreshPeriods.map(([period, label]) =>
              <option key={period} value={period}>{label}</option>
            )}
          </Select>
        </SelectAction>
        <SelectAction>
          <Select name="size" value={selectedSize} onChange={onChangeSize}>
            {refreshSizes.map(([size, label]) =>
              <option key={size} value={size}>{label}</option>
            )}
          </Select>
        </SelectAction>
      </SectionActions>
      <SectionHeader>
        <Button variant="primary" onClick={download}>
          <NxFontAwesomeIcon icon={faDownload}/>
          <span>{UIStrings.LOG_VIEWER.DOWNLOAD}</span>
        </Button>
        <Textfield
            className="nxrm-log-viewer-mark-text"
            name="mark"
            value={mark}
            onChange={updateMark}
            onKeyPress={onMarkKeyPress}
            placeholder={UIStrings.LOG_VIEWER.MARK_PLACEHOLDER}
        />
        <Button onClick={insertMark} id="insertMark">
          <NxFontAwesomeIcon icon={faMapMarker}/>
          <span>{UIStrings.LOG_VIEWER.INSERT_MARK}</span>
        </Button>
      </SectionHeader>
      <Textarea ref={textarea} readOnly wrap="false" value={logText}/>
    </Section>
  </ContentBody>;
}

function SelectAction({className, children, ...attrs}) {
  const classes = classNames('nxrm-select-action', className);

  return <div className={classes} {...attrs}>
    {children}
  </div>;
}
