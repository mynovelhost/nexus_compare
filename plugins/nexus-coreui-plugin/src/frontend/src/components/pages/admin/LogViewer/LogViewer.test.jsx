/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import { act } from 'react-dom/test-utils';
import {fireEvent, render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestUtils from 'nexus-ui-plugin/src/frontend/src/interface/TestUtils';
import axios from 'axios';

import LogViewer from './LogViewer';

jest.mock('axios', () => ({
  ...jest.requireActual('axios'), // Use most functions from actual axios
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn()
}));

describe('LogViewer', function() {
  const renderView = async () => {
    var selectors;
    await act(async () => {
      selectors = TestUtils.render(<LogViewer/>, ({queryByPlaceholderText}) => ({
        markInput: () => queryByPlaceholderText("Marker to insert into log")
      }));
    });
    return selectors;
  };

  const changeFieldAndAssertValue = async (fieldSelector, value) => {
    fireEvent.change(fieldSelector(), {target: {value: value}});
    await wait(() => expect(fieldSelector()).toHaveValue(value));
  };

  it('renders the log into the text area', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: 'log data'
    }));

    const {container} = await renderView();

    act(() => {
      expect(container.querySelector('textarea')).toHaveValue("log data");
    });

    expect(container).toMatchSnapshot();
  });

  it('creates a mark when insert mark clicked', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: 'log data'
    }));

    const {container, markInput, insertMarkButton} = await renderView();

    await changeFieldAndAssertValue(markInput, 'mymark');

    fireEvent.click(container.querySelector("#insertMark"));

    expect(axios.post).toBeCalled();
  });

  it('retrieves with specified size when selected', async function() {
    axios.get.mockReturnValue(Promise.resolve({
      data: 'log data'
    }));

    const {container} = await renderView();

    await changeFieldAndAssertValue(() => container.querySelector("select#size"), "100");

    expect(axios.get.mock.calls[1][1]).toMatchObject({params: {bytesCount: -102400}});
  });
});
