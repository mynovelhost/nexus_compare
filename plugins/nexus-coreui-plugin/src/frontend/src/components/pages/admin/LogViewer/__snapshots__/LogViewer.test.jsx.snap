// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogViewer renders the log into the text area 1`] = `
<div>
  <div
    class="nxrm-content-body nxrm-log-viewer"
  >
    <div
      class="nxrm-section nx-tile"
    >
      <div
        class="nxrm-section-actions nx-tile__actions"
      >
        <label
          for="period"
        >
          Refresh Interval
        </label>
        <div
          class="nxrm-select-action"
        >
          <select
            class="nxrm-select"
            id="period"
            name="period"
          >
            <option
              value="0"
            >
              Manual
            </option>
            <option
              value="20"
            >
              Every 20 seconds
            </option>
            <option
              value="60"
            >
              Every minute
            </option>
            <option
              value="120"
            >
              Every 2 minutes
            </option>
            <option
              value="300"
            >
              Every 5 minutes
            </option>
          </select>
        </div>
        <div
          class="nxrm-select-action"
        >
          <select
            class="nxrm-select"
            id="size"
            name="size"
          >
            <option
              value="25"
            >
              Last 25KB
            </option>
            <option
              value="50"
            >
              Last 50KB
            </option>
            <option
              value="100"
            >
              Last 100KB
            </option>
          </select>
        </div>
      </div>
      <div
        class="nxrm-section-header nx-tile-header"
      >
        <button
          class="nx-btn nx-btn--primary"
          type="input"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-download fa-w-16 nx-icon"
            data-icon="download"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
              fill="currentColor"
            />
          </svg>
          <span>
            Download
          </span>
        </button>
        <input
          class="nxrm-textfield nxrm-log-viewer-mark-text"
          id="mark"
          name="mark"
          placeholder="Marker to insert into log"
          type="text"
          value=""
        />
        <button
          class="nx-btn"
          id="insertMark"
          type="input"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-map-marker fa-w-12 nx-icon"
            data-icon="map-marker"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            Insert Mark
          </span>
        </button>
      </div>
      <textarea
        class="nxrm-textarea"
        readonly=""
        wrap="false"
      >
        log data
      </textarea>
    </div>
  </div>
</div>
`;
